frontend-react/src/pages/submissions/SubmissionDetails.tsx

```diff
@@ -0,0 +1,148 @@
+import { Suspense } from "react";
+import { useParams } from "react-router-dom";
+import { NetworkErrorBoundary, useResource } from "rest-hooks";
+
+import { getStoredOrg } from "../../components/GlobalContextProvider";
+import Spinner from "../../components/Spinner";
+import Title from "../../components/Title";
+import ActionDetailsResource, {
+    Destination,
+} from "../../resources/ActionDetailsResource";
+import { generateSubmissionDate } from "../../utils/DateTimeUtils";
+import { ErrorPage } from "../error/ErrorPage";
+
+/* Custom types */
+type DetailItemProps = {
+    item: string;
+    content: any;
+    subItem?: boolean;
+};
+
+type DestinationItemProps = {
+    destinationObj: Destination;
+};
+
+type SubmissionDetailsProps = {
+    actionId: string | undefined;
+};
+
+/* 
+    A component displaying a soft gray title and content in
+    standard black text. 
+
+    @param item - the title of a property; e.g. Report ID
+    @param content - the content of a property; e.g. 000000-0000-0000-000000
+*/
+export function DetailItem({ item, content, subItem }: DetailItemProps) {
+    return (
+        <div
+            style={{
+                display: "flex",
+                flexDirection: "column",
+                margin: subItem ? "16px 32px" : "8px 0px",
+            }}
+        >
+            <span>{item}</span>
+            <span>{content}</span>
+        </div>
+    );
+}
+
+/* 
+    A component displaying information about a single destination
+    returned from the history/submissions details API
+
+    @param destinationObj - a single object from the destinations array
+    in the history/submissions details API
+*/
+export function DestinationItem({ destinationObj }: DestinationItemProps) {
+    const submissionDate =
+        generateSubmissionDate(destinationObj.sending_at) || "unsent";
```

Comment:
i agree with LGTM bot here:
are we expecting `undefined` or `null` ever to return from `destinationObj.sending_at`? it looks like that method always returns an object? does `generateSubmissionDate` handle those conditions?

Reply:
I altered `generateSubmissionDate` to return null if the input couldn't parse to a date properly, and fixed up this file to account for a null date. :) Pushing soon.